### ISSUE TITLE
Mediasite with multiple video sources (added example with no authentication)

### ISSUE THREAD

**Issue by th3picow** [ytdl-org/youtube-dl#20611]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.04.07*. If it's not, read [this FAQ entry](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2019.04.07**

### Before submitting an *issue* make sure you have:
- [X] At least skimmed through the [README](https://github.com/ytdl-org/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/ytdl-org/youtube-dl#faq) and [BUGS](https://github.com/ytdl-org/youtube-dl#bugs) sections
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [X] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [X] Bug report (encountered problems with youtube-dl)
- [X] Feature request (request for a new functionality)

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Video: https://lecturecapture.wustl.edu/mediasite/Play/c63b969f11ff44c88b10249d51a86a2d1d

---

### Description of your *issue*, suggested solution and other information

There are two different sources of video in a "video", would it be possible to add a feature that would download them both instead of just one of them?

**Comment by fstirlitz** [comment]
```
$ youtube-dl -F 'https://lecturecapture.wustl.edu/mediasite/Play/c63b969f11ff44c88b10249d51a86a2d1d'
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading webpage
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading JSON metadata
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
[info] Available formats for c63b969f11ff44c88b10249d51a86a2d1d:
format code              extension  resolution note
video2-1.6-2_A_audio_1   m4a        audio only DASH audio  126k , m4a_dash container, mp4a.40.2 (44100Hz)
video2-1.2-2_A_audio_10  m4a        audio only DASH audio  128k , m4a_dash container, mp4a.40.2 (44100Hz)
video1-0.6-2_A_audio_1   m4a        audio only DASH audio  126k , m4a_dash container, mp4a.40.2 (44100Hz)
video1-0.2-2_A_audio_5   m4a        audio only DASH audio  128k , m4a_dash container, mp4a.40.2 (44100Hz)
video2-1.0-video-296     ismv       400x224     296k , AVC1, video only
video2-1.2-1_V_video_9   mp4        400x224    DASH video  296k , mp4_dash container, avc1.42400d, video only
video2-1.0-video-496     ismv       400x224     496k , AVC1, video only
video2-1.2-1_V_video_8   mp4        400x224    DASH video  496k , mp4_dash container, avc1.42400d, video only
video2-1.0-video-796     ismv       640x360     796k , AVC1, video only
video2-1.2-1_V_video_7   mp4        640x360    DASH video  796k , mp4_dash container, avc1.42401e, video only
video2-1.0-video-1296    ismv       640x360    1296k , AVC1, video only
video2-1.2-1_V_video_6   mp4        640x360    DASH video 1296k , mp4_dash container, avc1.4d401e, video only
video2-1.4-video-1498    ismv       960x540    1498k , AVC1, video only
video2-1.6-1_V_video_2   mp4        960x540    DASH video 1498k , mp4_dash container, avc1.4d001f, video only
video2-1.0-video-1596    ismv       960x540    1596k , AVC1, video only
video2-1.2-1_V_video_5   mp4        960x540    DASH video 1596k , mp4_dash container, avc1.4d401f, video only
video2-1.0-video-2596    ismv       960x540    2596k , AVC1, video only
video2-1.2-1_V_video_4   mp4        960x540    DASH video 2596k , mp4_dash container, avc1.4d401f, video only
video2-1.0-video-3313    ismv       1280x720   3313k , AVC1, video only
video2-1.2-1_V_video_3   mp4        1280x720   DASH video 3313k , mp4_dash container, avc1.4d401f, video only
video2-1.0-video-4596    ismv       1280x720   4596k , AVC1, video only
video2-1.2-1_V_video_2   mp4        1280x720   DASH video 4596k , mp4_dash container, avc1.4d4020, video only
video2-1.0-video-5596    ismv       1920x1080  5596k , AVC1, video only
video2-1.2-1_V_video_1   mp4        1920x1080  DASH video 5596k , mp4_dash container, avc1.4d4028, video only
video1-0.0-video-296     ismv       400x224     296k , AVC1, video only
video1-0.2-1_V_video_4   mp4        400x224    DASH video  296k , mp4_dash container, avc1.42400d, video only
video1-0.0-video-496     ismv       400x224     496k , AVC1, video only
video1-0.2-1_V_video_3   mp4        400x224    DASH video  496k , mp4_dash container, avc1.42400d, video only
video1-0.0-video-796     ismv       640x360     796k , AVC1, video only
video1-0.2-1_V_video_2   mp4        640x360    DASH video  796k , mp4_dash container, avc1.42401e, video only
video1-0.0-video-1296    ismv       640x360    1296k , AVC1, video only
video1-0.2-1_V_video_1   mp4        640x360    DASH video 1296k , mp4_dash container, avc1.4d401e, video only
video1-0.4-video-1499    ismv       960x540    1499k , AVC1, video only
video1-0.6-1_V_video_2   mp4        960x540    DASH video 1499k , mp4_dash container, avc1.4d001f, video only
video2-1.1               m3u8       unknown    
video2-1.5               m3u8       unknown    
video2-1.3               mp4        unknown    
video1-0.1               m3u8       unknown    
video1-0.5               m3u8       unknown    
video1-0.3               mp4        unknown    (best)
$ youtube-dl -f 'video1-0.3,video2-1.3' -o '%(title)s.%(format_id)s.%(ext)s' 'https://lecturecapture.wustl.edu/mediasite/Play/c63b969f11ff44c88b10249d51a86a2d1d'
```

This will download multiplex 0 of type 'video1' as one file, and multiplex 1 of type 'video2' as another. I used pre-multiplexed streams above for simplicity, but separate audio and video streams can be combined with the `+` operator as usual (e.g. `-f 'video2-1.2-1_V_video_1+video2-1.2-2_A_audio_10'`). If you want all streams in a single file, you'd have to wait for #6454 to get merged. I wouldn't hold my breath.

**Comment by th3picow** [comment]
> 
> 
> ```
> $ youtube-dl -F 'https://lecturecapture.wustl.edu/mediasite/Play/c63b969f11ff44c88b10249d51a86a2d1d'
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading webpage
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading JSON metadata
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
> WARNING: [Mediasite]  is not a supported codec
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
> WARNING: [Mediasite]  is not a supported codec
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
> WARNING: [Mediasite]  is not a supported codec
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading ISM manifest
> WARNING: [Mediasite]  is not a supported codec
> [Mediasite] c63b969f11ff44c88b10249d51a86a2d1d: Downloading MPD manifest
> [info] Available formats for c63b969f11ff44c88b10249d51a86a2d1d:
> format code              extension  resolution note
> video2-1.6-2_A_audio_1   m4a        audio only DASH audio  126k , m4a_dash container, mp4a.40.2 (44100Hz)
> video2-1.2-2_A_audio_10  m4a        audio only DASH audio  128k , m4a_dash container, mp4a.40.2 (44100Hz)
> video1-0.6-2_A_audio_1   m4a        audio only DASH audio  126k , m4a_dash container, mp4a.40.2 (44100Hz)
> video1-0.2-2_A_audio_5   m4a        audio only DASH audio  128k , m4a_dash container, mp4a.40.2 (44100Hz)
> video2-1.0-video-296     ismv       400x224     296k , AVC1, video only
> video2-1.2-1_V_video_9   mp4        400x224    DASH video  296k , mp4_dash container, avc1.42400d, video only
> video2-1.0-video-496     ismv       400x224     496k , AVC1, video only
> video2-1.2-1_V_video_8   mp4        400x224    DASH video  496k , mp4_dash container, avc1.42400d, video only
> video2-1.0-video-796     ismv       640x360     796k , AVC1, video only
> video2-1.2-1_V_video_7   mp4        640x360    DASH video  796k , mp4_dash container, avc1.42401e, video only
> video2-1.0-video-1296    ismv       640x360    1296k , AVC1, video only
> video2-1.2-1_V_video_6   mp4        640x360    DASH video 1296k , mp4_dash container, avc1.4d401e, video only
> video2-1.4-video-1498    ismv       960x540    1498k , AVC1, video only
> video2-1.6-1_V_video_2   mp4        960x540    DASH video 1498k , mp4_dash container, avc1.4d001f, video only
> video2-1.0-video-1596    ismv       960x540    1596k , AVC1, video only
> video2-1.2-1_V_video_5   mp4        960x540    DASH video 1596k , mp4_dash container, avc1.4d401f, video only
> video2-1.0-video-2596    ismv       960x540    2596k , AVC1, video only
> video2-1.2-1_V_video_4   mp4        960x540    DASH video 2596k , mp4_dash container, avc1.4d401f, video only
> video2-1.0-video-3313    ismv       1280x720   3313k , AVC1, video only
> video2-1.2-1_V_video_3   mp4        1280x720   DASH video 3313k , mp4_dash container, avc1.4d401f, video only
> video2-1.0-video-4596    ismv       1280x720   4596k , AVC1, video only
> video2-1.2-1_V_video_2   mp4        1280x720   DASH video 4596k , mp4_dash container, avc1.4d4020, video only
> video2-1.0-video-5596    ismv       1920x1080  5596k , AVC1, video only
> video2-1.2-1_V_video_1   mp4        1920x1080  DASH video 5596k , mp4_dash container, avc1.4d4028, video only
> video1-0.0-video-296     ismv       400x224     296k , AVC1, video only
> video1-0.2-1_V_video_4   mp4        400x224    DASH video  296k , mp4_dash container, avc1.42400d, video only
> video1-0.0-video-496     ismv       400x224     496k , AVC1, video only
> video1-0.2-1_V_video_3   mp4        400x224    DASH video  496k , mp4_dash container, avc1.42400d, video only
> video1-0.0-video-796     ismv       640x360     796k , AVC1, video only
> video1-0.2-1_V_video_2   mp4        640x360    DASH video  796k , mp4_dash container, avc1.42401e, video only
> video1-0.0-video-1296    ismv       640x360    1296k , AVC1, video only
> video1-0.2-1_V_video_1   mp4        640x360    DASH video 1296k , mp4_dash container, avc1.4d401e, video only
> video1-0.4-video-1499    ismv       960x540    1499k , AVC1, video only
> video1-0.6-1_V_video_2   mp4        960x540    DASH video 1499k , mp4_dash container, avc1.4d001f, video only
> video2-1.1               m3u8       unknown    
> video2-1.5               m3u8       unknown    
> video2-1.3               mp4        unknown    
> video1-0.1               m3u8       unknown    
> video1-0.5               m3u8       unknown    
> video1-0.3               mp4        unknown    (best)
> $ youtube-dl -f 'video1-0.3,video2-1.3' -o '%(title)s.%(format_id)s.%(ext)s' 'https://lecturecapture.wustl.edu/mediasite/Play/c63b969f11ff44c88b10249d51a86a2d1d'
> ```
> 
> This will download multiplex 0 of type 'video1' as one file, and multiplex 1 of type 'video2' as another. I used pre-multiplexed streams above for simplicity, but separate audio and video streams can be combined with the `+` operator as usual (e.g. `-f 'video2-1.2-1_V_video_1+video2-1.2-2_A_audio_10'`). If you want all streams in a single file, you'd have to wait for #6454 to get merged. I wouldn't hold my breath.

Damn, youtube-dl keeps amazing me with the ways you can use it, cheers wil give that a try

**Comment by th3picow** [comment]
``
youtube-dl https://mediasite.uis.no/Mediasite/Catalog/catalogs/sta100-v2019 --cookies=cookies.txt -f video1-0.3,video3-2.3 -o %(title)s.%(format_id)s.%(ext)s -i
[MediasiteNamedCatalog] sta100-v2019: Downloading webpage
[MediasiteCatalog] f7ecebfa-3b45-4307-ab9f-8dad7b7123f8: Downloading webpage
[MediasiteCatalog] f7ecebfa-3b45-4307-ab9f-8dad7b7123f8: Downloading JSON metadata
[download] Downloading playlist: STA100 - våren 2019
[MediasiteCatalog] playlist STA100 - våren 2019: Collected 39 video ids (downloading 39 of them)
--
[download] Downloading video 3 of 39
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading webpage
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading JSON metadata
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading MPD manifest
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading MPD manifest
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading MPD manifest
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading ISM manifest
WARNING: [Mediasite]  is not a supported codec
[Mediasite] 5bb389e121b04b1f989c8f426e386f9f1d: Downloading MPD manifest
[info] 5bb389e121b04b1f989c8f426e386f9f1d: downloading video in 2 formats
[download] Resuming download at byte 322960384
[download] Destination: STA100 tirsdag_4_9_2019.video1-0.3.mp4
[download] 100% of 2.41GiB in 06:18
[download] Destination: STA100 tirsdag_4_9_2019.video3-2.3.mp4
[download]  53.1% of 348.65MiB at  3.58MiB/s ETA 00:45
``
That solved it for me, thank you made my life so much easier
